### PR TITLE
Monitor manually added agents

### LINKED ISSUE #22: Monitor: allow to add agents by hand
Sometimes zeroconf may be turned off. Adding agents by hand should be possible

### PR DESCRIPTION
**Pull request by Kicer86**

### CODE REVIEW

File: src/monitor/ManualAgentsValidator.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
+#include "ManualAgentsValidator.hpp"
+#include "AgentInformation.hpp"
+
+
+void ManualAgentsValidator::addNewAgent(const QString& name, const QString& ip, const QString& port)
```
Comment (wkucjas) [inline src/monitor/ManualAgentsValidator.cpp:5]: This is an option which allow user to add agent without zeroconf service installed or this is for something else?

File: src/monitor/ManualAgentsValidator.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    const bool ipValid = std::all_of(segments.begin(), segments.end(), [](const QString& segment) {
+        const int number = segment.toInt();
+        return number >= 0 && number <= 255;
+    });
+
+    const int portNum = port.toInt();
+
+    if (ipValid && portNum >= 0 && portNum <= 65535)
+    {
+        AgentInformation info(name, ip, portNum, AgentInformation::DetectionSource::Hardcoded);
```
Comment (wkucjas) [inline src/monitor/ManualAgentsValidator.cpp]: Maybe before adding procedure monitor should ask agent is it up and running? GET message with "status" or something?

**Review by wkucjas — COMMENTED** [review]
Please answer the questions.

File: src/monitor/ManualAgentsValidator.cpp (hunk 1 above)
Comment (Kicer86) [reply src/monitor/ManualAgentsValidator.cpp:5]: "This is an option which allow user to add agent without zeroconf service installed" yes

**Review by wkucjas — APPROVED** [review]

**Pull request merged**